web/src/search/input/toggles/CopyQueryButton.tsx

```diff
@@ -0,0 +1,42 @@
+import copy from 'copy-to-clipboard'
+import ContentCopyIcon from 'mdi-react/ContentCopyIcon'
+import { Tooltip } from '../../../components/tooltip/Tooltip'
+import React, { useCallback } from 'react'
+import classNames from 'classnames'
+import { Observable, merge, of, timer } from 'rxjs'
+import { tap, switchMapTo, startWith } from 'rxjs/operators'
+import { useEventObservable } from '../../../../../shared/src/util/useObservable'
+
+interface Props {
+    fullQuery: string
+    className?: string
+}
+
+/**
+ * A repository header action that copies the current page's URL to the clipboard.
+ */
+export const CopyQueryButton: React.FunctionComponent<Props> = (props: Props) => {
+    const [nextClick, copied] = useEventObservable(
+        useCallback(
+            (clicks: Observable<React.MouseEvent>) =>
+                clicks.pipe(
+                    tap(() => copy(props.fullQuery)),
+                    switchMapTo(merge(of(true), timer(1000).pipe(switchMapTo(of(false))))),
```

Comment:
```suggestion
                    switchMapTo(merge(of(true), timer(1000).pipe(mapTo(false)))),
```

Reply:
Actually, I think this is wrong because `timer()` emits infinitely. Should be
```suggestion
                    switchMapTo(merge(of(true), of(false).pipe(delay(1000))),
```

Reply:
If no `periodOrScheduler` argument is passed to `timer()`, it emits one value then completes. The `switchMapTo(of())` was dumb though 🤦